katas/args/daniel/Parser.js

```diff
@@ -0,0 +1,43 @@
+export class Parser {
+    parse( args, schema ) {
+        let parsedArgs = [];
+        if ( schema.validateSchema(args.getArgsList()) ) {
+            
+            args.getArgsList().forEach(unparsedArg => {
+                parsedArgs.push( this.parseArg(unparsedArg, schema) );
+            });
+            
+        } else {
+            throw new Error('The argument does not exist in the schema');
+        }
+
+        return parsedArgs;
+    }
+
+    parseArg( unparsedArg, schema ) {
+        const flag = unparsedArg.flag;
+        let value = unparsedArg.value;
+        let parsedArg = {};
+
+        if ( value == '' ) {
+            parsedArg = {flag: flag, value: schema.getPropertyOfArg( flag, 'defaultValue')};
+        } else {
+            switch ( schema.getPropertyOfArg(flag, 'type') ) {
+            case 'int':
```

Comment:
As discussed, this could be improved by creating some sort of `Flag` class that could contain the type information and return the correct type